app/models/concerns/foreman_ansible/config_report_extensions.rb

```diff
@@ -0,0 +1,35 @@
+module ForemanAnsible
+  module ConfigReportExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      class << self
+        alias_method :orig_import, :import
+
+        def import(report, proxy_id = nil)
```

Comment:
I think the logic should be in code and plugins should only provide ways how to identify report origin